mk/config.mk

```diff
@@ -35,11 +35,15 @@ WARNS ?= 3
 # Define NOWERROR=1 so that warnings are not treated as errors
 # NOWERROR=1
 
-# Define DEBUG=1 to compile with -g option
+# Define DEBUG=1 to compile without optimization (forces -O0)
 # DEBUG=1
 
-# If 1, debug mode of the tee firmware (CPU restart, Core Status)
-CFG_TEE_CORE_DEBUG ?= 0
+# If y, enable debug mode of the tee firmware (CPU restart, Core Status
+# verbose, panic & assert verbose). When disable, NDEBUG directive is defined.
+CFG_TEE_CORE_DEBUG ?= n
+ifneq ($(CFG_TEE_CORE_DEBUG),y)
+NDEBUG ?= 1
```

Comment:
@etienne-lms NACK for testing `CFG_TEE_CORE_DEBUG` in `assert.h`, it's just too ugly ;)

I think we were (I was) on the wrong track trying to make `NDEBUG` a configuration variable in `conf.mk` and `conf.h`, because as @jenswi-linaro  pointed out, it cannot be global (has to be different when building core and TA libs).
So... what about this:
```diff
diff --git a/core/core.mk b/core/core.mk
index c44d1c1..1f7f43a 100644
--- a/core/core.mk
+++ b/core/core.mk
@@ -16,6 +16,9 @@ COMPILER_$(sm)                ?= $(COMPILER)
 include mk/$(COMPILER_$(sm)).mk
 
 cppflags$(sm)  += -D__KERNEL__
+ifneq ($(CFG_TEE_CORE_DEBUG),y)
+cppflags$(sm)  += -DNDEBUG
+endif
 
 cppflags$(sm)  += -Icore/include
 cppflags$(sm)  += -include $(conf-file)
```